cleanlab/object_detection/summary.py

```diff
@@ -444,6 +449,138 @@ def visualize(
+        tpfp = pool.starmap(
+            _calculate_true_positives_false_positives,
+            zip(
+                pred_bboxes,
+                lab_bboxes,
+                [iou_threshold for _ in range(num_images)],
+                [True for _ in range(num_images)],
+            ),
+        )
+        for j, tpfp_j in enumerate(tpfp):
+            for k, tpfp_k in enumerate(tpfp_j):
+                counter_dict[class_num][k] += np.sum(tpfp_k)
+
+    results: DefaultDict[int, Dict[str, int]] = collections.defaultdict(dict)
+    for class_num in counter_dict:
+        results[class_num]["TP"] = counter_dict[class_num][0]
+        results[class_num]["FP"] = counter_dict[class_num][1]
+        results[class_num]["FN"] = counter_dict[class_num][2]
+    return results
+
+
+def get_average_per_class_confusion_matrix_(
+    labels: List[Dict[str, Any]], predictions: List[np.ndarray], num_procs: int = 1
+) -> Dict[int, Dict[str, float]]:
+    """
+    Parameters
+    ----------
+    labels:
+        A list of ``N`` dictionaries such that ``labels[i]`` contains the given labels for the `i`-th image.
+        Refer to documentation for this argument in :py:func:`find_label_issues <cleanlab.object_detection.filter.find_label_issues>` for further details.
+
+    predictions:
+        A list of ``N`` ``np.ndarray`` such that ``predictions[i]`` corresponds to the model predictions for the `i`-th image.
+        Refer to documentation for this argument in :py:func:`find_label_issues <cleanlab.object_detection.filter.find_label_issues>` for further details.
+
+    num_procs:
+        Number of processes for parallelization. Default is 1.
+
+    Returns
+    -------
+        Average of TP,FP, FN for each class across iou thresholds.
+    """
+    iou_thrs = np.linspace(0.5, 0.95, int(np.round((0.95 - 0.5) / 0.05)) + 1, endpoint=True)
+    num_classes = len(predictions[0])
+    avg_metrics = {class_num: {"TP": 0.0, "FP": 0.0, "FN": 0.0} for class_num in range(num_classes)}
+
+    for iou_threshold in iou_thrs:
+        results_dict = _get_per_class_confusion_matrix_dict_(
+            labels, predictions, iou_threshold, num_procs
+        )
+
+        for class_num in results_dict:
+            tp = results_dict[class_num]["TP"]
+            fp = results_dict[class_num]["FP"]
+            fn = results_dict[class_num]["FN"]
+
+            avg_metrics[class_num]["TP"] += tp
+            avg_metrics[class_num]["FP"] += fp
+            avg_metrics[class_num]["FN"] += fn
+
+    num_thresholds = len(iou_thrs) * len(results_dict)
+
+    for class_num in avg_metrics:
+        avg_metrics[class_num]["TP"] /= num_thresholds
+        avg_metrics[class_num]["FP"] /= num_thresholds
+        avg_metrics[class_num]["FN"] /= num_thresholds
+    return avg_metrics
+
+
+def calculate_per_class_metrics(
+    labels: List[Dict[str, Any]], predictions: List[np.ndarray], num_procs: int = 1
+) -> Dict[int, Dict[str, float]]:
+    """
```

Comment:
missing short description of what this method is supposed to do. This is a public-facing method and needs clear description of its purpose

Reply:
you need to make it clear that a high-level this is about evaluating the trained object detection model.

Most of the methods in cleanlab are about assessing the data itself. So when we have methods for assessing models, it needs to be clearly stated this method is for assessing models, not data